### PR TITLE
feat(cli,ironfish): Add `wallet:multisig:sign`

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Adds a CLI command to aggregate signature shares from participants

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    let account = flags.account
+    if (!account) {
+      const response = await client.wallet.getDefaultAccount()
+
+      if (!response.content.account) {
+        this.error(
+          `No account is currently active.
+           Use ironfish wallet:create <name> to first create an account`,
+        )
```
Comment (hughy) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:70]: should we say something to indicate that the user needs to import the right multisig account?

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+export class MultiSigSign extends IronfishCommand {
+  static description = 'Aggregate signature shares from participants to sign a transaction'
+  static hidden = true
+
+  static flags = {
+    ...RemoteFlags,
+    account: Flags.string({
+      description: 'The account that created the raw transaction',
+      required: false,
+    }),
```
Comment (patnir) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:18]: should we have an `f` char like `wallet:send`?

File: ironfish-cli/src/commands/wallet/multisig/sign.ts (hunk 1 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/wallet/multisig/sign.ts:70]: Yep, good idea. Fixed

File: ironfish-cli/src/commands/wallet/multisig/sign.ts (hunk 2 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/wallet/multisig/sign.ts]: Fixed

**Review by hughy — APPROVED** [review]

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { RemoteFlags } from '../../../flags'
+
+export class MultiSigSign extends IronfishCommand {
+  static description = 'Aggregate signature shares from participants to sign a transaction'
+  static hidden = true
+
+  static flags = {
+    ...RemoteFlags,
+    account: Flags.string({
+      char: 'f',
```
Comment (jowparks) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:16]: do we use `-f` for other commands? I have been using `-a`

Comment (jowparks) [reply ironfish-cli/src/commands/wallet/multisig/sign.ts:16]: Looks like `-f` is better, we should change it to that

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    if (!signatureShares) {
+      const input = await CliUx.ux.prompt('Enter the signature shares separated by commas', {
+        required: true,
+      })
+      signatureShares = input.split(',')
+    }
+    signatureShares = signatureShares.map((s) => s.trim())
+
+    const client = await this.sdk.connectRpc()
+
```
Comment (jowparks) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:61]: Should we just handle this in RPC with an optional account and try to use default account if not passed?

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    CliUx.ux.action.stop()
+
+    if (response.content.accepted === false) {
+      this.warn(
+        `Transaction '${transaction.hash().toString('hex')}' was not accepted into the mempool`,
+      )
+    }
+
+    if (response.content.broadcasted === false) {
```
Comment (jowparks) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:94]: Should this be `flags.broadcasted && response.content.broadcasted === false`?

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { CurrencyUtils, Transaction } from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import { IronfishCommand } from '../../../command'
+import { RemoteFlags } from '../../../flags'
+
+export class MultiSigSign extends IronfishCommand {
```
Comment (jowparks) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:9]: `Multisig` now

File: ironfish-cli/src/commands/wallet/multisig/sign.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    }),
+    signatureShare: Flags.string({
+      char: 's',
+      description: 'Participant signature share',
+      multiple: true,
+    }),
+    broadcast: Flags.boolean({
+      default: true,
+      allowNo: true,
+      description: 'Broadcast the transaction after signing',
```
Comment (jowparks) [inline ironfish-cli/src/commands/wallet/multisig/sign.ts:36]: nit: broadcast to the ironfish network

File: ironfish-cli/src/commands/wallet/multisig/sign.ts (hunk 4 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/wallet/multisig/sign.ts:61]: Yeah we can move it there

File: ironfish-cli/src/commands/wallet/multisig/sign.ts (hunk 7 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/wallet/multisig/sign.ts]: fixed

**Pull request merged**